go-controller/pkg/ovn/pods.go

```diff
@@ -81,6 +81,16 @@ func (oc *Controller) deleteLogicalPort(pod *kapi.Pod) {
 	portInfo, err := oc.logicalPortCache.get(logicalPort)
 	if err != nil {
 		klog.Errorf(err.Error())
+		// Even if the port is not in the cache, IPs annotated in the Pod annotation may already be allocated,
```

Comment:
You already had another PR that was trying to clean up things when they were not in the cache. I had asked you to refactor that to pull the annotation and and attempt to delete things anyway:
https://github.com/ovn-org/ovn-kubernetes/pull/1557#pullrequestreview-466092852

Should we abandon the above and also handle logical port deletion here as well?

Reply:
With recent changes in addLogicalPort(), the logical switch port is created after the port is added into the port cache, so the problem fixed by PR 1557 no longer exists. I will close that PR.

Reply:
Correct. But I still see downstream even with that fix it happens. I think it might have something to do with restarting ovnkube master and the cache getting wiped and not being rebuilt correctly. I would appreciate if we catch that case anyway to ensure the ports are cleaned up.

Reply:
repushed, please take a look